SQL/TSQL.sublime-syntax

```diff
@@ -275,23 +283,34 @@ contexts:
     - match: (?=\S)
       pop: true
 
+  table-alias:
+    - meta_prepend: true
+    - include: with
+    - match: \(
+      scope: punctuation.section.group.begin.sql
+      push: inside-with-group
+
   after-table-alias:
     - meta_prepend: true
     - include: with
+    - include: with-paren
 
   with:
     - match: (?i)\bwith\b
       scope: keyword.other.DML.sql
-      push: with-paren
+      push: with-paren-or-pop
 
   with-paren:
     - match: \(
       scope: punctuation.section.group.begin.sql
       set: inside-with-group
     - match: \b(?i:nowait)\b
       scope: keyword.other.tsql
+
+  with-paren-or-pop:
+    - include: with-paren
     - match: (?=\S)
-      pop: true
+      set: table-alias
```

Comment:
I'm not sure exactly what's going on, but `WITH cte_table AS ( SELECT ... )` does weird stuff. It thinks the `(` is an identifier, not the opening of a sub-select. Also see `WITH cte_table (col_name) AS ( SELECT ... )`.

Reply:
I think I have fixed this now :)